### PR TITLE
Create/Edit Gradeables saves in real time (mostly)

### PR DESCRIPTION
**Pull request by KevinMackenzie**
The new create/edit gradeables interface now pushes updates to the database every time a field is changed.  A status message informs the user of how this process is going, and any issues will show up as tool tips for the fields with errors.  These tool tips are not enabled for any of the rubrics, but the numeric/text and checkpoint rubrics do update in real time.  The only part of the form that does not is the electronic file rubric, which must be updated with the press of a 'save' button.

Other changes include:
*various style changes to reduce clutter and improve readability
*Improved server-side validation of submitted data
*naming conventions for class properties better enforced

### CODE REVIEW

**Comment by KevinMackenzie** [comment]
Other than the create/edit gradeable page, there have been changes to the gradeable config generation and some name changes in the autograding files.  These are very minor changes, but have the potential to break autograding completely, so that should be tested.

**Comment by KevinMackenzie** [comment]
@holzbh You've been able to get weird date consistency behavior on the previous forms.  Can you try the new one out and see if you can get any weirdness?

For now, there are a couple of hidden div's that contain debugging information at the top of the form if you get any weird persistent errors. (`ajax-debug` and `ajax-raw`)

**Comment by bmcutler** [comment]
@KevinMackenzie please resolve merge conflilcts

**Comment by holzbh** [comment]
I don't like not having an edit button if the gradeable is in a "Run BUILD_sample.sh" mode - I often edit the gradeable to go see what the ID was so I don't have to rebuild all gradeables. Not sure if it's because what's happening below.

Made a new gradeable (thissisawstj), I can see the .json in config/form correctly appears. I just used the default settings for an electronic gradeable, so it's using one of the simple python tests.
>root@vagrant:/var/local/submitty/courses/s18/sample# ls config/form/form_thissisawstj.json -l
>-rw-r----- 1 hwphp sample_tas_www 203 Jun 12 19:39 config/form/form_thissisawstj.json

Running BUILD_sample.sh
>root@vagrant:/var/local/submitty/courses/s18/sample# ./BUILD_sample.sh thissisawstj
>BUILDING course=sample semester=s18...
>Tue Jun 12 19:50:45 EDT 2018
>WARNING: thissisawstj is not a valid electronic gradeable

As a result, I wasn't able to go back in and change the date, no edit button for me.

The json I see is:

```
{
    "g_id": "thissisawstj",
    "eg_submission_due_date": {
        "date": "2018-06-19 23:59:59.000000",
        "timezone_type": 1,
        "timezone": "-04:00"
    },
    "eg_is_repository": false
}
```

**Comment by KevinMackenzie** [comment]
The last commit I pushed should fix the issues with the build script `BUILD_sample.sh` file failing, but I never had an issue with the edit button disappearing.  If you do still have the edit button disappear, can you post the row from the `gradeable` table for that gradeable?

**Comment by holzbh** [comment]
I don't have that state anymore, but I'm testing your branch now* so I'll see if I can reproduce.

**Comment by KevinMackenzie** [comment]
Electronic rubric marks do not load if `Use TA Grading` is off on page load.

**Comment by holzbh** [comment]
Build works fine now, Edit button appears, and dates seem to correctly be checked.

The red highlighting of the box is fine (along with the "Some Changes Failed") error. I think we could do more visually, but that's another discussion for another day. I'm happy with the functionality we have now.

**Comment by bmcutler** [comment]
@MasterOdin @KevinMackenzie 
I'm merging this PR now, but I would like Kevin to address Matt's concerns in a later PR.

File: site/app/controllers/admin/AdminGradeableController.php [hunk 1]
```diff
@@ -47,318 +58,552 @@ public function run() {
...
+
+    private static function error($data)
+    {
+        return [0, $data];
+    }
+
+    private static function warning($data)
+    {
+        return [1, $data];
+    }
```
[the remark this answers is not in the document]
Comment (KevinMackenzie) [reply site/app/controllers/admin/AdminGradeableController.php]: Warnings are supposed to be messages that the user sees, but that don't prevent the model from saving.  This is a server-side workaround for one edge case that should just be done on the client-side.  I came to this realization when using some of the validation code from this class in the new gradeable model.

**Comment by KevinMackenzie** [comment]
Making the properties public was a naive convenience.  I was unaware how the magic methods could be manipulated.  The date validation code from this PR was copied into the other PR, so that's why its the same.  For the DB queries.  I didn't mess with what they did, I just reformatted them (and changed them to use the public properties instead of the magic methods).  All of these changes will be made as a part of porting the `AdminGradeableController` to use the new `Gradeable` model.

**Pull request merged**